src/Microsoft.NetCore.Analyzers/Core/Security/UseAutoValidateAntiforgeryToken.cs

```diff
@@ -0,0 +1,333 @@
+                                }
+                            }
+                        }
+                    }
+                }, SymbolKind.NamedType);
+
+                compilationStartAnalysisContext.RegisterCompilationEndAction(
+                (CompilationAnalysisContext compilationAnalysisContext) =>
+                {
+                    if (usingValidateAntiForgeryAttribute && !hasGlobalAntiForgeryFilter && actionMethodSymbols.Any())
+                    {
+                        var visited = new HashSet<ISymbol>();
+                        var results = new Dictionary<ISymbol, HashSet<ISymbol>>();
+
+                        if (onAuthorizationAsyncMethodSymbols.Any())
+                        {
+                            foreach (var calleeMethod in inverseGraph.Keys)
+                            {
+                                if (calleeMethod.Name == "ValidateRequestAsync" &&
+                                    (calleeMethod.ContainingType.AllInterfaces.Contains(iAntiforgeryTypeSymbol) ||
+                                    calleeMethod.ContainingType.Equals(iAntiforgeryTypeSymbol)))
+                                {
+                                    FindAllTheSpecifiedCalleeMethods(calleeMethod, visited, results);
+
+                                    if (results.Values.Any(s => s.Any()))
+                                    {
+                                        return;
+                                    }
+                                }
+                            }
+                        }
+
+                        foreach (var (methodSymbol, attributeName) in actionMethodSymbols)
+                        {
+                            compilationAnalysisContext.ReportDiagnostic(
+                                methodSymbol.CreateDiagnostic(
+                                    Rule,
+                                    methodSymbol.Name,
+                                    attributeName));
+                        }
+                    }
+                });
+
+                // <summary>
+                // Analyze the method to find all the specified method it calls, in this case, all the method symbols in onAuthorizationAsyncMethodSymbols.
+                // </summary>
+                // <param name="methodSymbol">The symbol of the method to be analyzed</param>
+                // <param name="visited">All the method has been analyzed</param>
+                // <param name="results">The result is organized by &lt;method to be analyzed, dangerous method it calls&gt;</param>
+                void FindAllTheSpecifiedCalleeMethods(ISymbol methodSymbol, HashSet<ISymbol> visited, Dictionary<ISymbol, HashSet<ISymbol>> results)
+                {
+                    if (visited.Add(methodSymbol))
+                    {
+                        results.Add(methodSymbol, new HashSet<ISymbol>());
+
+                        if (!inverseGraph.TryGetValue(methodSymbol, out var calledMethods))
```

Comment:
>calledMethods [](start = 76, length = 13)

Nit: Since this is an inverseGraph, it's tracking callingMethods, not calledMethods, right?